### PR TITLE
Update codeblock style and add border-radius settings

### PR DESCRIPTION
**Pull request by sli1989**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [X] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [X] Tests for the changes have been added (for bug fixes / features).
   - [ ] Muse | Mist have been tested.
   - [ ] Pisces | Gemini have been tested.
- [X] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [X] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

- Optimize the visual style of codeblock
- Add the option of border-radius of codeblock. if `codeblock.rounded=false`, the codeblock would be loaded with rounded corner.

Issue Number(s): N/A

## What is the new behavior?

`codeblock.rounded=true`

* Screens with this changes: N/A
* Link to demo site with this changes: http://test.saili.science/posts/codeblock-test/

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [ ] No.

### CODE REVIEW

File: source/css/_common/components/highlight/highlight.styl [hunk 1]
```diff
@@ -64,6 +66,10 @@ pre {
     color: $highlight-foreground;
     line-height: 1em;
     margin-bottom: 1em;
+	margin: 0em;
+	padding: 0.5em;
+	background: #eee;
```
Comment (ivan-nginx) [inline source/css/_common/components/highlight/highlight.styl]: https://github.com/theme-next/hexo-theme-next/blob/52d3c024e2e8cb3f1a538ed780187ec8d944f2ed/source/css/_variables/base.styl#L12

Comment (sli1989) [reply source/css/_common/components/highlight/highlight.styl]: anything else?

Comment (sli1989) [reply source/css/_common/components/highlight/highlight.styl]: to set `border-radius: ?+px`? i don't know the css grammar. 😂

Comment (ivan-nginx) [reply source/css/_common/components/highlight/highlight.styl]: Yep. By this variable we can set custom border radius too. See `sidebar.offset` in 3rd example.

> i don't know the css grammar. 

This actually like JS grammar. This is swig. Try it as in examples, it's not hard.

Comment (sli1989) [reply source/css/_common/components/highlight/highlight.styl]: yeah, I saw it..

**Comment by ivan-nginx** [comment]
But, here is some else thing...

**Comment by sli1989** [comment]
I have deleted the `pre` element in last commit. the latest commit is fine.

**Comment by ivan-nginx** [comment]
> the latest commit is fine.

Demo with latest commit & border radius?

**Comment by sli1989** [comment]
The latest living demo has changed to test footer icon... Above feature hasn't merged.

File: source/css/_common/components/highlight/highlight.styl [hunk 2]
```diff
@@ -22,14 +22,12 @@ code {
...
   background: $code-background;
   border-radius: $code-border-radius;
-  font-size $code-font-size;
+  font-size: $code-font-size;
 }
 
 pre {
   @extend $code-block;
   padding: 10px;
-  if hexo-config('codeblock.rounded') { border-radius: 5px;}
```
Comment (sli1989) [inline source/css/_common/components/highlight/highlight.styl]: Here has deleted.

**Comment by ivan-nginx** [comment]
So, merge icon and go see here?

**Comment by sli1989** [comment]
Live demo updated. There is no problem.

**Comment by ivan-nginx** [comment]
So, yeah. It's good for now.

**Pull request merged**